internal/tools/aws/database_perseus.go

```diff
@@ -0,0 +1,1251 @@
+	}
+
+	return multitenantDatabases, nil
+}
+
+// getAssignedMultitenantDatabaseResources returns the assigned multitenant
+// database if there is one or nil if there is not. An error is returned if the
+// installation is assigned to more than one database.
+func (d *PerseusDatabase) getAndLockAssignedProxiedDatabase(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) (*model.DatabaseResourceGrouping, func(), error) {
+	multitenantDatabases, err := store.GetMultitenantDatabases(&model.MultitenantDatabaseFilter{
+		DatabaseType:          d.databaseType,
+		InstallationID:        d.installationID,
+		MaxInstallationsLimit: model.NoInstallationsLimit,
+		Paging:                model.AllPagesNotDeleted(),
+	})
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to query for multitenant databases")
+	}
+	if len(multitenantDatabases) > 1 {
+		return nil, nil, errors.Errorf("expected no more than 1 assigned database for installation, but found %d", len(multitenantDatabases))
+	}
+	if len(multitenantDatabases) == 0 {
+		return nil, nil, nil
+	}
+
+	unlockFn, err := lockMultitenantDatabase(multitenantDatabases[0].ID, d.instanceID, store, logger)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to lock multitenant database")
+	}
+	databaseResources, err := store.GetOrCreateProxyDatabaseResourcesForInstallation(d.installationID, multitenantDatabases[0].ID)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to get database resources")
+	}
+
+	return databaseResources, unlockFn, nil
+}
+
+func (d *PerseusDatabase) provisionPerseusDatabase(vpcID string, rdsCluster *rdsTypes.DBCluster, logger log.FieldLogger) error {
+	rdsID := *rdsCluster.DBClusterIdentifier
+
+	logger.Infof("Provisioning Perseus database %s", rdsID)
+
+	masterDBSecret, err := d.client.Service().secretsManager.GetSecretValue(
+		context.TODO(),
+		&secretsmanager.GetSecretValueInput{
+			SecretId: rdsCluster.DBClusterIdentifier,
+		})
+	if err != nil {
+		return errors.Wrapf(err, "failed to find the master secret for the multitenant proxy cluster %s", rdsID)
+	}
+
+	perseusUser, err := d.ensurePerseusDatabaseUserSecretIsCreated(&rdsID, &vpcID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to ensure the perseus database user secret was created for %s", rdsID)
+	}
+
+	pgbouncerUser, err := d.ensurePGBouncerAuthUserSecretIsCreated(&rdsID, &vpcID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to ensure the pgbouncer auth user secret was created for %s", rdsID)
+	}
+
+	close, err := d.connectToRDSCluster(rdsPostgresDefaultSchema, *rdsCluster.Endpoint, DefaultMattermostDatabaseUsername, *masterDBSecret.SecretString)
+	if err != nil {
+		return errors.Wrapf(err, "failed to connect to the multitenant proxy cluster %s", rdsID)
+	}
+	defer close(logger)
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(DefaultMySQLContextTimeSeconds*time.Second))
```

Comment:
`DefaultMySQLContextTimeSeconds` ?

Reply:
Yeah, it's more of a general timeout that we use, but let's create a new one to clear up naming confusion. 👍